tools/build-tools/src/common/fluidRepo.ts

```diff
@@ -37,44 +27,39 @@ export class FluidRepo {
     public readonly packages: Packages;
     constructor(public readonly resolvedRoot: string, services: boolean) {
         const packageManifest = getPackageManifest(resolvedRoot);
-        this.clientMonoRepo = new MonoRepo(MonoRepoKind.Client, this.resolvedRoot);
-        if (packageManifest.serverPath) {
-            this.serverMonoRepo = new MonoRepo(MonoRepoKind.Server, path.join(this.resolvedRoot, packageManifest.serverPath));
-        }
 
-        let additionalPackages: Package[] = [];
-
-        packageManifest.repoPackages.client.forEach((fluidPackage: IFluidRepoPackage) => {
-            additionalPackages = [
-                ...additionalPackages,
-                ...Packages.loadDir(path.join(resolvedRoot, fluidPackage.directory), undefined, fluidPackage.ignoredDirs)
-            ]
-        });
-
-        if (services) {
-            assert(packageManifest.repoPackages.server.optional, "Requested optional server packages without passing parameters in package.json");
-            packageManifest.repoPackages.server.optional!.forEach((fluidPackage: IFluidRepoPackage) => {
-                additionalPackages = [
-                    ...additionalPackages,
-                    ...Packages.loadDir(path.join(resolvedRoot, fluidPackage.directory), undefined, fluidPackage.ignoredDirs)
-                ]
-            });
-        } else {
-            packageManifest.repoPackages.server.required.forEach((fluidPackage: IFluidRepoPackage) => {
-                additionalPackages = [
-                    ...additionalPackages,
-                    ...Packages.loadDir(path.join(resolvedRoot, fluidPackage.directory), undefined, fluidPackage.ignoredDirs)
-                ]
-            });
+        const loadOneEntry = (item: string | IFluidRepoPackage, group: string) => {
+            if (typeof item === "string") {
+                return Packages.loadDir(path.join(resolvedRoot, item), group);
+            }
+            return Packages.loadDir(path.join(resolvedRoot, item.directory), group, undefined, item.ignoredDirs);
+        }
+        const loadedPackages: Package[] = [];
+        let clientMonoRepo: MonoRepo | undefined;
+        for (const group in packageManifest.repoPackages) {
+            const item = packageManifest.repoPackages[group];
+            if (group === "client") {
```

Comment:
Please see comment in package.json to avoid having to hardcode this group string check here.
I think having repoPackages.client and repoPackage.server from the type definition also lines up with the fact that we specifically look for a MonoRepoKind.Client and MonoRepoKind.Server here.
The other packages can go under services perhaps?